pythonarcade/arcade
# Add Sound page to the Programming Guide

Closes #1274
Closes #1900

### Changes

1. Add link targets to the platformer tutorial
2. Add new Sound page to the programming guide covering usage and compatibility
3. Link the Sound page from the top-level index
 
### How to Test

1. Clone the branch locally
2. Run `pip install -I -e .[dev]` to make sure you have the dev dependencies fully installed & upgraded
3. `./make.py html` (optional, but helps prevent an occasional rebuild loop)
4. `./make.py serve`
5. Open https://localhost:8000 in your browser
6. Verify the page is in the sidebar and Programming Guide homepage section
7. Open http://localhost:8000/programming_guide/sound.html
8. Proof-read the Sound Basics sections brutally:
    * http://localhost:8000/programming_guide/sound.html#sound-basics-loading
    * http://localhost:8000/programming_guide/sound.html#sound-basics-playing
    * http://localhost:8000/programming_guide/sound.html#sound-basics-stopping
9. Verify that [#1900](https://github.com/pythonarcade/arcade/issues/1900) is covered by http://localhost:8000/programming_guide/sound.html#backends-determine-playback-features 
10. If time / interest allow, proof read and verify other parts

### Follow-up work for future PRs

1. Move some of the preloading concepts to a new performance page / heading
2. Add an easier to digest reference counting section somewhere and link it instead of the Python contributor doc
3. Move some compatibility information to its own page / heading
